src/components/Assignment.js

```diff
@@ -13,14 +18,56 @@ import {SERVER_URL} from '../constants.js'
 //
 
 class Assignment extends React.Component {
-    constructor(props) {
-      super(props);
-      this.state = {selected: 0, assignments: []};
-    };
+  constructor(props) {
+    super(props);
+    this.state = {selected: 0, assignments: [], open: false};
+  };
  
-   componentDidMount() {
+  componentDidMount() {
     this.fetchAssignments();
   }
+
+  handleClickOpen = () => {
+    this.setState({ open:true });
+  };
+
+  handleClose = () => {
+    this.setState({ open:false });
+  };
+
+  handleChange = (event) => {
+    this.setState({[event.target.name]: event.target.value});
+  }
+
+  handleSubmit = () => {
+    this.addAssignment(this.state.assignmentName, this.state.dueDate, this.state.courseId);
+    this.setState({ open:false });
+  }
+
+  addAssignment = (name, dueDate, courseId) => {
+    console.log("Assignment.addAssignment");
+    const token = Cookies.get('XSRF-TOKEN');
+    fetch(`${SERVER_URL}/course/${courseId}/add`, 
+      {  
+        method: 'POST', 
+        headers: { 
+          'X-XSRF-TOKEN': token,
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({assignmentName: name, dueDate: dueDate})
+      } )
+    .then((response) => { 
```

Comment:
It might not be necessary, but you could put a toast.success here for when the assignment is successfully added

Reply:
Added